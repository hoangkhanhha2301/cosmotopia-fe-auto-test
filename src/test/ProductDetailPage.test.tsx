import { render, screen } from '@testing-library/react';
import ProductDetailPage from '@/test/pages/ProductDetailPage';
import userEvent from '@testing-library/user-event';

const product = {
  id: '1',
  name: 'Áo sơ mi',
  price: 150000,
  description: 'Chất liệu cotton, thoáng mát.',
};

describe('ProductDetailPage', () => {
  it('hiển thị thông tin chi tiết sản phẩm', () => {
    const addToCart = jest.fn();
    render(<ProductDetailPage product={product} addToCart={addToCart} />);
    expect(screen.getByText(/áo sơ mi/i)).toBeInTheDocument();
    expect(screen.getByText(/150000/i)).toBeInTheDocument();
    expect(screen.getByText(/cotton/i)).toBeInTheDocument();
  });

  it('hiển thị tên, giá và mô tả chi tiết đầy đủ', () => {
    const addToCart = jest.fn();
    render(<ProductDetailPage product={product} addToCart={addToCart} />);

    expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('Áo sơ mi');
    expect(screen.getByText(/giá: 150000 VND/i)).toBeInTheDocument();
    expect(screen.getByText(/chất liệu cotton, thoáng mát/i)).toBeInTheDocument();
   });

  it('hiển thị đúng đơn vị tiền tệ VND', () => {
    const addToCart = jest.fn();
    render(<ProductDetailPage product={product} addToCart={addToCart} />);
    const priceText = screen.getByText(/giá:/i).textContent;
    expect(priceText).toMatch(/150000.*VND/);
  });

  it('hiển thị thông báo nếu mô tả trống', () => {
    const noDescProduct = { ...product, description: '' };
    const addToCart = jest.fn();
    render(<ProductDetailPage product={noDescProduct} addToCart={addToCart} />);

    expect(screen.getByText(/không có mô tả/i)).toBeInTheDocument();
  });

  it('khớp với giao diện hiện tại (snapshot)', () => {
    const addToCart = jest.fn();
    const { container } = render(<ProductDetailPage product={product} addToCart={addToCart} />);
    expect(container).toMatchSnapshot();
  });

  it('gọi callback khi click Thêm vào giỏ hàng', async () => {
    const addToCart = jest.fn();
    render(<ProductDetailPage product={product} addToCart={addToCart} />);

    await userEvent.click(screen.getByText(/thêm vào giỏ hàng/i));
    expect(addToCart).toHaveBeenCalledWith('1');
  });


});
