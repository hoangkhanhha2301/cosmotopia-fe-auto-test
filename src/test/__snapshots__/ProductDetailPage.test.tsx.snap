// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetailPage khớp với giao diện hiện tại (snapshot) 1`] = `
<div>
  <div>
    <h1>
      Áo sơ mi
    </h1>
    <p>
      Giá: 
      150000
       VND
    </p>
    <p>
      Chất liệu cotton, thoáng mát.
    </p>
    <button>
      Thêm vào giỏ hàng
    </button>
  </div>
</div>
`;
